### PR TITLE
improve(hardhat): add kovan network and improve bridge script

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

Minor improvements were needed to complete bridge war games.

**Summary**

- Updated the bridge task to compute the resource id.
- Added the kovan network to the hardhat config.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

File: packages/common/src/hardhat/tasks/bridge.js [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
     };
 
+    const _getResourceId = (name, chainId) => {
+      const encodedParams = web3.eth.abi.encodeParameters(["string", "uint8"], [name, chainId]);
+      return web3.utils.soliditySha3(encodedParams);
+    };
+
+    if (rname || cid) {
+      if (id) throw new Error("Cannot provide --cid or --resouceName with --id");
+      if (!cid || !rname) throw new Error("Must provide --cid and --resouceName when generating the resouce id");
```
Comment (nicholaspai) [inline packages/common/src/hardhat/tasks/bridge.js:40]:
```suggestion
      if (!cid || !rname) throw new Error("Must provide --cid and --resouceName when generating the resource id");
```

File: packages/common/src/hardhat/tasks/bridge.js [hunk 2]
```diff
@@ -1,27 +1,48 @@
...
 const BLANK_FUNCTION_SIG = "0x00000000";
 // Admin role on Bridge contract:
 const DEFAULT_ADMIN_ROLE = "0x0000000000000000000000000000000000000000000000000000000000000000";
 
 task("register-generic-resource", "Admin can set generic resource ID on Bridge")
-  .addParam("id", "Resource ID to set", "0x0", types.string)
-  .addParam("target", "Contract to delegate call to for this resource ID", ZERO_ADDRESS, types.string)
+  .addParam("target", "Contract to delegate call to for this resource ID", undefined, types.string)
+  .addOptionalParam("rname", "Resouce name to generate Resource ID", undefined, types.string)
+  .addOptionalParam("cid", "Chain ID to generate Resource ID", undefined, types.int)
```
Comment (nicholaspai) [inline packages/common/src/hardhat/tasks/bridge.js:11]: This is a cool feature. So example call might be to use `--rname Oracle --cid 1` for any contract communicating with the Source on mainnet?

**Review by nicholaspai — APPROVED** [review]
LGTM after you respond to my nits

File: packages/common/src/hardhat/tasks/bridge.js (hunk 2 above)
Comment (mrice32) [reply packages/common/src/hardhat/tasks/bridge.js:11]: Exactly. I thought forcing them to compute the resource id was cumbersome.

**Comment by mrice32** [comment]
@nicholaspai I messed up doing some find and replace, so the misspelled flags aren't even the correct flag names anymore. Will update shortly.

File: packages/common/src/hardhat/tasks/bridge.js (hunk 2 above)
Comment (nicholaspai) [reply packages/common/src/hardhat/tasks/bridge.js:11]: cool lgtm

**Pull request merged**